lib/WasmReader/WasmBinaryReader.cpp

```diff
@@ -72,6 +72,17 @@ WasmBinaryReader::ReadFromModule()
 
     SectionCode sectionId;
     UINT length = 0;
+
+    if (ReadConst<UINT32>() != 0x6d736100)
```

Comment:
Good catch. Since the SExprParser doesn't read any header, I'm using a bit of state to ensure checking this only once now. Having a separate ReadModuleHeader() would have been nicer, but let's consider that later.